avm/res/network/express-route-circuit/main.bicep

```diff
@@ -0,0 +1,289 @@
+@secure()
+@description('Optional. The shared key for peering configuration. Router does MD5 hash comparison to validate the packets sent by BGP connection. This parameter is optional and can be removed from peering configuration if not required.')
+param sharedKey string = ''
+
+@description('Optional. The autonomous system number of the customer/connectivity provider.')
+param peerASN int = 0
+
+@description('Optional. A /30 subnet used to configure IP addresses for interfaces on Link1.')
+param primaryPeerAddressPrefix string = ''
+
+@description('Optional. A /30 subnet used to configure IP addresses for interfaces on Link2.')
+param secondaryPeerAddressPrefix string = ''
+
+@description('Optional. Specifies the identifier that is used to identify the customer.')
+param vlanId int = 0
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Allow classic operations. You can connect to virtual networks in the classic deployment model by setting allowClassicOperations to true.')
+param allowClassicOperations bool = false
+
+@description('Optional. The bandwidth of the circuit when the circuit is provisioned on an ExpressRoutePort resource. Available when configuring Express Route Direct. Default value of 0 will set the property to null.')
+param bandwidthInGbps int = 0
+
+@description('Optional. The reference to the ExpressRoutePort resource when the circuit is provisioned on an ExpressRoutePort resource. Available when configuring Express Route Direct.')
+param expressRoutePortResourceId string = ''
+
+@description('Optional. Flag denoting global reach status. To enable ExpressRoute Global Reach between different geopolitical regions, your circuits must be Premium SKU.')
+param globalReachEnabled bool = false
+
+@description('Optional. The diagnostic settings of the service.')
+param diagnosticSettings diagnosticSettingType
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
+
+var peeringConfiguration = [
+  {
+    name: peeringType
+    properties: {
+      peeringType: peeringType
+      sharedKey: sharedKey
+      peerASN: peerASN
+      primaryPeerAddressPrefix: primaryPeerAddressPrefix
+      secondaryPeerAddressPrefix: secondaryPeerAddressPrefix
+      vlanId: vlanId
+    }
+  }
+]
```

Comment:
As this cariable is only used in one location, I'd suggest to move it into the actual resource deployment below. You should be able to replace it 1:1

Reply:
Note, if changed may require a regeneration of the `main.json` file